chain-indexer/src/domain/storage.rs

```diff
@@ -29,7 +30,14 @@ where
     async fn get_contract_action_count(&self) -> Result<(u64, u64, u64), sqlx::Error>;
 
     /// Save the given [Block], update transaction IDs, and return the max transaction ID.
-    async fn save_block(&self, block: &mut Block) -> Result<Option<u64>, sqlx::Error>;
+    /// The `network_id` parameter is required for contract balance extraction during indexing.
```

Comment:
Notice that serialization is being reworked and the NetworkId will be removed. Just FYI. Which makes me happy, because (de)serialization should not be a storage concern.